internal/controllers/app_controller.go

```diff
@@ -173,6 +173,29 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App) reconci
 			message: err.Error(),
 		}
 	}
+
+	// check for canary deployment
+	if app.Status.IsActiveCanary {
+		if app.Status.CurrentCanaryStep != app.Spec.Canary.Steps {
+			if *app.Spec.Canary.NextScheduledTime == metav1.Now() {
```

Comment:
I think we shouldn't use `==` there. 
Does it work? 
What do you think of `app.Spec.Canary.NextScheduledTime <= metav1.Now()`?